lib/rex/version.rb

```diff
@@ -0,0 +1,12 @@
+
+class Rex::Version < Gem::Version
+
+  def initialize(version)
+    if version == nil
+      # Rubygems 4 is deprecating `nil` as a valid version number
+      # Currently it is the equivalent of a `0` so we set that here to keep the same functionality
+      version = 0
+    end
+    super version
+  end
+end
```

Comment:
💎 👮‍♂️ 👀

Reply:
I'll rebase this and give it a once over with rubocop soon after I'm done with my current task

Reply:
done :)